src/auth/digest/eDirectory/ldap_backend.cc

```diff
@@ -204,17 +205,21 @@ getpassword(char *login, char *realm)
     char *password = NULL;
     int retry = 0;
     char filter[8192];
+    *filter = '\0';
     char searchbase[8192];
+    *searchbase = '\0';
     char *universal_password = NULL;
     size_t universal_password_len = 256;
     int nmas_res = 0;
     int rc = -1;
     if (ld) {
         if (usersearchfilter) {
             char escaped_login[1024];
-            snprintf(searchbase, sizeof(searchbase), "%s", userbasedn);
+            std::snprintf(searchbase, sizeof(searchbase), "%s", userbasedn);
+            searchbase[sizeof(searchbase)-1] = '\0';
```

Comment:
Here and elsewhere in the PR, I think we should not add termination when using std::snprintf() that is guaranteed to terminate and, I presume, has no known cases where that guarantee is broken. Avoiding noisy and error-prone explicit termination is one of the biggest selling points of using std::snprintf() instead of its less standardized and known-to-be-broken-in-some-environments variants!

```suggestion
```

Removing PR-added termination lines will also make the modified snprintf() callers more consistent with each other because other callers modified in this PR do not terminate.

You are the first author switching to std::snprintf(), so there may be surprises we do not know about, but I think we should assume standard-compliant STL implementations until proven otherwise (at least). Otherwise, we are likely to increase code inconsistencies while getting lost in second-guessing standard functions and ourselves.